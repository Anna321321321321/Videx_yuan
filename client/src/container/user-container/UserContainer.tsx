import React, { Component } from 'react';
import { connect } from 'react-redux';
import Avatar from '../../components/avatar';
import * as UserStore from '../../stores/user-store';

interface UserContainerProps {
  id: string;
  name: string;
  email: string;
  type: number;
  initialized: boolean;
  init: Function;
}

export class UserContainer extends Component<UserContainerProps, any> {
  componentDidMount() {
    if (!this.props.initialized) {
      this.props.init();
    }
  }

  generateInitial() {
    const name: string = this.props.name;
    const firstName = name.split(' ')[0];
    const lastName = name.split(' ')[1];
    return (firstName ? firstName[0] : '') + (lastName ? lastName[0] : '');
  }

  render() {
    if (!this.props.initialized) {
      return null;
    } else {
      return (
        <div>
          <Avatar name={this.generateInitial()} />
        </div>
      );
    }
  }
}

const mapStateToProps = state => UserStore.selectors.flattenData(state);

const mapDispatchToProps = {
  init: UserStore.actions.init
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(UserContainer);
