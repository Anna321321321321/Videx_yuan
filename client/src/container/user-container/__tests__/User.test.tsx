import { mount, shallow } from 'enzyme';
import React from 'react';
import { Provider } from 'react-redux';
import { UserContainer } from '../UserContainer';

describe('User Component Unit Tests', () => {
  it('Initialize component', () => {
    const mockInit = jest.fn();
    const wrapper = shallow(
      <UserContainer
        id={null}
        name={null}
        email={null}
        type={null}
        initialized={false}
        init={mockInit}
      />
    );
    expect(mockInit.mock.calls.length).toBe(1);
    expect(wrapper).toMatchSnapshot();
  });

  it('After initialized', () => {
    const mockInit = jest.fn();
    const wrapper = shallow(
      <UserContainer
        id="46bf1e47-bd1f-40a8-b7b3-8f8781c04077"
        name="Teacher One"
        email="teacher1@videxint.onmicrosoft.com"
        type={1}
        initialized={true}
        init={mockInit}
      />
    );
    expect(mockInit.mock.calls.length).toBe(0);
    expect(wrapper).toMatchSnapshot();
  });
});
