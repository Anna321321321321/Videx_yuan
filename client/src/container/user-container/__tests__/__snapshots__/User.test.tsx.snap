// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Component Unit Tests After initialized 1`] = `
<div>
  <Persona
    name="TO"
  />
</div>
`;

exports[`User Component Unit Tests Initialize component 1`] = `""`;
